RPT_Application/src/main/java/com/harman/web_pages/UserReportCRUDView.java

```diff
@@ -0,0 +1,171 @@
+
+	private H3 wwbH3;
+
+	public UserReportCRUDView(ReportService reportService, EmployeeService employeeService) {
+		super(reportService, employeeService);
+
+		super.getGrid().addItemDoubleClickListener(e -> taskDialog(super.getGrid(), e, super.getEmployee()));
+
+		Button addButton = new Button("Add Report");
+		addButton.setWidth("250px");
+		addButton.setHeight("75px");
+
+		addButton.addClickListener(e -> createSaveDialog(super.getGrid(), super.getEmployee()));
+		add(addButton);
+	}
+
+	private void taskDialog(Grid<Report> grid, ItemDoubleClickEvent<Report> e, Employee employee) {
+		Dialog dialog = new Dialog();
+		HorizontalLayout layout = new HorizontalLayout();
+		FormLayout buttomLayout = new FormLayout();
+
+		layout.setJustifyContentMode(JustifyContentMode.BETWEEN);
+		layout.add(new Label("Month: " + e.getItem().getDate()));
+		Icon exitIcon = new Icon(VaadinIcon.CLOSE);
+		exitIcon.getStyle().set("cursor", "pointer");
+		exitIcon.setColor("red");
+		exitIcon.addClickListener((__) -> dialog.close());
+		layout.add(exitIcon);
+
+		Button addTaskButton = new Button("Add Task");
+		addTaskButton.setHeight("75px");
+		addTaskButton.setWidth("250px");
+
+		addTaskButton.addClickListener(event -> {
+			createTask(TaskCRUDView.getGrid(), e.getItem(), employee);
+		});
+
+		wwbH3 = new H3();
+		wwbH3.setText("WWB: " + e.getItem().getWwb());
+		buttomLayout.add(addTaskButton, wwbH3);
+
+		TaskCRUDView taskCRUDView = new TaskCRUDView(taskService, e.getItem());
+		dialog.add(layout, taskCRUDView, buttomLayout);// , addTaskButton, wwbTextArea);
+		taskCRUDView.setHeight("500px");
+		dialog.setHeight("600px");
+		dialog.setWidth("1000px");
+		dialog.open();
+	}
+
+	private void createSaveDialog(Grid<Report> grid, Employee employee) {
+		Dialog dialog = new Dialog();
+
+		DatePicker datePicker = new DatePicker();
+		datePicker.setPlaceholder("Choose date");
+
+		Button saveButton = new Button("Save");
+		saveButton.addClickListener(a -> {
+			Report report = new Report();
+			report.setDate(datePicker.getValue());
+			report.setEmployee(employee);
+			employee.addReport(report);
+			reportService.save(report);
+			// employeeService.save(employee);
+
+			dialog.close();
+			grid.setItems(reportService.findByEmployeeUsername(employee.getUsername()));
+		});
+
+		dialog.add(datePicker, saveButton);
+		dialog.open();
+	}
+
+	private void createTask(Grid<Task> grid, Report report, Employee employee) {
+		Dialog dialog = new Dialog();
+		FormLayout formLayout = new FormLayout();
+
+		dialog.setHeightFull();
+		dialog.setWidth("750px");
+
+		TextArea typeTextArea = new TextArea("Type");
+		typeTextArea.setSizeFull();
+		TextArea nameTextArea = new TextArea("Name");
+		nameTextArea.setSizeFull();
+		TextArea placementTextArea = new TextArea("Placement");
+		placementTextArea.setSizeFull();
+		TextArea descriptionTextArea = new TextArea("Description");
+		descriptionTextArea.setSizeFull();
+
+		ListBox<String> listBox = new ListBox<>();
```

Comment:
We can use enum here to avoid string -> enum conversion.

Reply:
that's much better